### PR TITLE
Feature/26 admin merchant update

### LINKED ISSUE #32: feature/26-admin-merchant-update
26. Admin Merchant Update

As an admin,
When I visit a merchant's admin show page (/admin/merchants/:merchant_id)
Then I see a link to update the merchant's information.
When I click the link
Then I am taken to a page to edit this merchant
And I see a form filled in with the existing merchant attribute information
When I update the information in the form and I click ‘submit’
Then I am redirected back to the merchant's admin show page where I see the updated information
And I see a flash message stating that the information has been successfully updated.

### PR DESCRIPTION
**Pull request by aplasket**
## Checklist before requesting a review
- [x] Verified Render deploy is functional
- [x] Pulled/merged main 
- [x] Tests for the changes have been added into spec files
- [x] Branch name follows our guidelines 
- [x] All tests running clean
- [x] Ran this in Rails S  - features are working properly
- [x] Test coverage is at 100% for models and features

## What changes or new behavior does this PR introduce (Bug fix, feature, update, etc)?
    - added update action to /merchants controller
    - added edit form_with model and action to admin/merchants
    - updated routes for admin to have access to merchant update
    - updated show page with correct update button
   
## Bugs, Issues, etc:
 - [ ] check if there are known bugs or issue

### CODE REVIEW

File: app/views/admin/merchants/edit.html.erb
Comment (AlexKiselich) [inline app/views/admin/merchants/edit.html.erb:1]: This was cool!

File: app/views/layouts/application.html.erb
Comment (AlexKiselich) [inline app/views/layouts/application.html.erb:1]: This is also very cool!

**Review by AlexKiselich — APPROVED** [review]
Great work!

File: app/views/admin/merchants/edit.html.erb
Comment (MylesNottingham) [inline app/views/admin/merchants/edit.html.erb:1]: Nice!

File: app/controllers/merchants_controller.rb [hunk 1]
```diff
@@ -3,4 +3,19 @@ def show
...
+    if params[:merchant][:name]
+      if merchant.update(merchant_params)
+        redirect_to admin_merchant_path(merchant)
+        flash[:success] = "Merchant has been successfully updated"
+      end
+    end
+  end
+
+  private
+  def merchant_params
```
Comment (MylesNottingham) [inline app/controllers/merchants_controller.rb]: I think it it convention to have a blank line above and below "private" but I could be wrong.

File: app/views/admin/merchants/show.html.erb [hunk 2]
```diff
@@ -1 +1,2 @@
-<%= @merchant.name %>
\ No newline at end of file
+<h2><%= @merchant.name %></h2>
+<p><%= link_to "Update #{@merchant.name}", "/admin/merchants/#{@merchant.id}/edit" %></p>
```
Comment (MylesNottingham) [inline app/views/admin/merchants/show.html.erb]: Opportunity for path helper

File: spec/features/admin/merchants/show_spec.rb [hunk 3]
```diff
@@ -17,5 +20,18 @@
       expect(page).to have_content(merchant_2.name)
       expect(page).to_not have_content(merchant_1.name)
     end
+
+    it "has a button to update a merchant's info" do
+      merchant_1 = Merchant.create!(name: "GingerBread Maker")
+      merchant_2 = Merchant.create!(name: "CandyMan")
+      merchant_3 = Merchant.create!(name: "Toasty Toads")
+
+      visit "/admin/merchants/#{merchant_3.id}"
```
Comment (MylesNottingham) [inline spec/features/admin/merchants/show_spec.rb]: Opportunity for path helper

**Review by AlexKiselich — APPROVED** [review]

**Review by MylesNottingham — APPROVED** [review]
Excellent work on this! I learned a lot here.

**Pull request merged**